Expensify/expensify-common
# Fix problem with parsing links containing mailto: prefix followed by an email

This PR fixes potential problems with parsing links containing mailto: prefix followed by an email. The changes were added to enhance parsing inside Live Markdown Preview.
The problem was mainly with parsing `[mailto:someone@example.com](mailto:someone@example.com)`. ExpensiMark instead of one link returns two separate <a> tags inside brackets: 
```
[<a href=\"mailto:someone@example.com\">someone@example.com</a>](mailto:<a href=\"mailto:someone@example.com\">someone@example.com</a>)
```

To fix the problem we decided to collapse such links, similarly like it is done after writing 
`[someone@example.com](someone@example.com)`. 

So now following strings:
- `[someone@example.com](mailto:someone@example.com)`
- `[mailto:someone@example.com](someone@example.com)`
- `[mailto:someone@example.com](mailto:someone@example.com)`

will be collapsed into: `<a href=\"mailto:someone@example.com\">someone@example.com</a>`

https://github.com/Expensify/expensify-common/assets/39538890/c7ed7e97-94d9-4153-b1ae-7bd821c1e3de

### Fixed Issues
$ https://github.com/Expensify/App/issues/31181

# Tests
To test these changes I've added proper tests connected to all needed behaviors for Live Markdown Preview. To test it by yourself you should:
1. Run the NewDot/App
2. Open any chat
3. Send message that contains:
```
[someone@example.com](someone@example.com)
[someone@example.com](mailto:someone@example.com)
[mailto:someone@example.com](someone@example.com)
[mailto:someone@example.com](mailto:someone@example.com)
```
4. Verify if every string is collapsed into just someone@example.com

# QA
Same as tests
